problems/[230112] 48. Rotate Image.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * leetcode problem link: https://leetcode.com/problems/rotate-image/
+ *
+ * @param {number[][]} matrix
+ * @return {void} Do not return anything, modify matrix in-place instead.
+ */
+const rotate = function(matrix) {
+    
```

Comment:
```suggestion
    const rotate = function(matrix) {
      const tempMatrix = [...matrix];
      const rotateCopyMatrix = [];
      let index = 0;
  
      while (tempMatrix[0].length) {
          const tempMatrixItem = [];
  
          for (let item of tempMatrix) {
              tempMatrixItem.push(item.shift());
          }
          rotateCopyMatrix.push(tempMatrixItem.reverse());
      }
  
      while (index < matrix.length) {
          for (let i = 0; i < matrix.length; i++) {
              matrix[index][i] = rotateCopyMatrix[index][i];
          }
  
          index++;
      }
  };
```

은혜님 아까 말한대로 저 matrix copy해서 순서에 맞게 돌린 matrix를 만들어주고
matrix 원본을 순회하면서 순서대로 값을 할당해줬어요 
copy를 해줘가지구 공간 효율은 별로 안좋을 것 같긴하네요..!

Reply:
오... 저렇게 reverse 해서 할 수 있겠군요 현정님 감사해요🙇‍♀️